plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/apsadmin/resource/MultipleResourceAction.java

```diff
@@ -0,0 +1,340 @@
+                    }
+                }
+
+            } catch (Throwable ex) {
+                this.addFieldError(FILE_UPLOAD_FIELD, this.getText("error.resource.file.void"));
+            }
+
+            if (null != getFileUpload()) {
+                validateFileDescriptions();
+                if (null != this.getResourceType()) {
+                    ResourceInterface resourcePrototype = this.getResourceManager().createResourceType(this.getResourceType());
+
+                    this.getFileUploadFileName().forEach(fileName
+                            -> checkRightFileType(resourcePrototype, fileName));
+
+                } else {
+                    this.addFieldError(FILE_UPLOAD_FIELD, this.getText("error.resource.file.genericError"));
+                }
+            }
+        }
+    }
+
+    @Override
+    public String edit() {
+        try {
+            savedId.clear();
+            ResourceInterface resource = this.loadResource(this.getResourceId());
+            this.setResourceTypeCode(resource.getType());
+            List fileDescr = new ArrayList<String>();
+            fileDescr.add(resource.getDescription());
+            setFileDescriptions(fileDescr);
+            List<Category> resCategories = resource.getCategories();
+            for (int i = 0; i < resCategories.size(); i++) {
+                Category resCat = resCategories.get(i);
+                this.getCategoryCodes().add(resCat.getCode());
+            }
+            this.setMainGroup(resource.getMainGroup());
+            this.setStrutsAction(ApsAdminSystemConstants.EDIT);
+
+        } catch (Throwable t) {
+            logger.error("error in edit", t);
+            return FAILURE;
+        }
+        return SUCCESS;
+    }
+
+    protected void validateFileDescriptions() {
+        int fileUploadSize = 0;
+        if (null != this.getFileUpload()) {
+            fileUploadSize = this.getFileUpload().size();
+        }
+        for (int i = 0; i < fileUploadSize; i++) {
+            if (null != fileDescriptions) {
+                if (!(fileDescriptions.get(i).length() > 0)) {
+                    this.addFieldError(DESCR_FIELD + i, this.getText("error.resource.file.descrEmpty"));
+                }
+                if (fileDescriptions.get(i).length() > 250) {
+                    this.addFieldError(DESCR_FIELD + i, this.getText("error.resource.file.descrTooLong"));
+                }
+            } else {
+                this.addFieldError(DESCR_FIELD + i, this.getText("error.resource.file.descrEmpty"));
+            }
+        }
+    }
+
+    public String joinCategory() {
+        fetchFileDescriptions();
+        return super.joinCategory();
+    }
+
+    public String removeCategory() {
+        fetchFileDescriptions();
+        return super.removeCategory();
+    }
+
+    protected void checkRightFileType(ResourceInterface resourcePrototype, String fileName) {
```

Comment:
This method should have some logging. Especially in the case where the file type is invalid